src/main/java/svydovets/web/dto/ResponseEntity.java

```diff
@@ -0,0 +1,87 @@
+package svydovets.web.dto;
+
+import jakarta.annotation.Nullable;
+
+public class ResponseEntity<T>{
+
+    private final T body;
+    private HttpStatus status;
+    private HttpHeaders headers;
+
+    public ResponseEntity(T body) {
+        this.body = body;
+    }
+
+    public ResponseEntity(T body, HttpHeaders headers, HttpStatus status) {
+        this.body = body;
+        this.headers = headers;
+        this.status = status;
+    }
+
+    public T getBody() {
+        return body;
+    }
+
+    public HttpStatus getStatus() {
+        return status;
+    }
+
+    public HttpHeaders getHeaders() {
+        return headers;
+    }
+
+    public static BodyBuilder ok() {
+        return status(HttpStatus.OK);
+    }
+
+    public static BodyBuilder status(HttpStatus status) {
+        return new DefaultBuilder(status);
+    }
+
+    public interface HeadersBuilder{
+        BodyBuilder header(String headerName, String headerValues);
+    }
+
+    public interface BodyBuilder extends HeadersBuilder{
+        BodyBuilder contentLength(int length);
+        BodyBuilder contentType(String contentType);
+        <T> ResponseEntity<T> body(@Nullable T body);
+        <T> ResponseEntity<T> build();
+
+    }
+
+    private static class DefaultBuilder implements BodyBuilder {
```

Comment:
It may be better if "BodyBuilder" will not extends "HeadersBuilder" and class "DefaultBuilder" will implement 2 interfaces